src/OrleansTestingHost/TestCluster.cs

```diff
@@ -415,7 +421,8 @@ public void InitializeClient()
             GrainClient.Initialize(clientConfig);
             this.GrainFactory = GrainClient.GrainFactory;
             this.InternalGrainFactory = this.GrainFactory as IInternalGrainFactory;
-            this.StreamProviderManager = RuntimeClient.Current.CurrentStreamProviderManager;
+            this.ServiceProvider = GrainClient.ServiceProvider;
+            this.StreamProviderManager = this.ServiceProvider.GetRequiredService<IRuntimeClient>().CurrentStreamProviderManager;
```

Comment:
if it's a small change, sure, otherwise this is a good first step. No need to hold the merge for something like this, as this aspect is not worse than before (and the rest is all advantageous)